tests/acceptance/bashunit_upgrade_test.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+function set_up() {
+  ./build.sh >/dev/null
+  LATEST_VERSION="$(helpers::get_latest_tag)"
+  TEST_ENV_FILE="tests/acceptance/fixtures/.env.default"
+}
+
+function tear_down() {
+  rm -f ./bin/bashunit
+}
+
+function test_do_not_upgrade_when_latest() {
+  local output
+  output="$(./bin/bashunit --upgrade)"
+
+  assert_equals "> You are already on latest version" "$output"
+  assert_string_ends_with "$LATEST_VERSION" "$(./bin/bashunit --version --env "$TEST_ENV_FILE")"
+}
+
+function test_upgrade_when_a_new_version_found() {
+  sed -i -e \
+    's/declare -r BASHUNIT_VERSION="[0-9]\{1,\}\.[0-9]\{1,\}\.[0-9]\{1,\}"/declare -r BASHUNIT_VERSION="0.1.0"/' \
+    ./bin/bashunit
+
+  if [[ $_OS == "OSX" ]]; then
+    rm ./bin/bashunit-e
```

Comment:
`./bin/bashunit-e`, is that correct?

Reply:
Yes, it's correct. This is because `sed` behaves a it different in macOS than others. 
Concretely, the option `-e` is used for regex in linux; while in macOS it's not needed and then it's appending `-e` to the file: therefore it generates a new file named 'bashunit-e' 🤣 
To keep it simple, @Tito-Kati and I decided simply rm that file and we are good to go.